sdk/core/Azure.Core.TestFramework/src/ClientDiagnosticListener.cs

```diff
@@ -124,11 +134,14 @@ public void Dispose()
                 subscription.Dispose();
             }
 
-            foreach (ProducedDiagnosticScope producedDiagnosticScope in Scopes)
+            if (_collect.Value)
```

Comment:
Why don't you check this in the beginning of the method?

Reply:
You need to validate all collected scopes. We won't collect any scopes we don't care about.